core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1006,6 +1005,10 @@ private void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, Ex
 
             // Update log start offset with the computed value after retention cleanup is done
             remoteLogRetentionHandler.logStartOffset.ifPresent(offset -> handleLogStartOffsetUpdate(topicIdPartition.topicPartition(), offset));
```

Comment:
nit: Can we move Line 1007-1011 to 985? The `deleteLogSegmentsDueToLeaderEpochCacheTruncation` should run only for case when there is a truncation in the leader-epoch-checkpoint file.